src/memory/memoryManager.ts

```diff
@@ -33,13 +33,13 @@ export class MemorySegmentManager {
       }
     }
 
-    const segmentSize = UnsignedInteger.toUint64(BigInt(data.length));
+    const segmentSize = UnsignedInteger.toUint32(data.length);
```

Comment:
I think the segment size should be the sum of data.length and something,

Possibly: address.getOffset() + data.length
But what about memory holes?

Reply:
I don't think memory holes are relevant here right?